show/fgnhg.py

```diff
@@ -47,123 +41,154 @@ def active_hops(nhg):
+    try:
+        table_keys = sorted(state_db.keys(state_db.STATE_DB, _hash))
+    except Exception as e: 
+        ctx.fail("FG_ROUTE_TABLE does not exist!")
+    if table_keys is None:
+        ctx.fail("FG_ROUTE_TABLE does not exist!")
     if nhg is None:
         for nhg_prefix in table_keys:
             bank_dict = {}
             t_dict = state_db.get_all(state_db.STATE_DB, nhg_prefix)
             vals = sorted(set([val for val in t_dict.values()]))
-
             for nh_ip in vals:
                 bank_ids = sorted([int(k) for k, v in t_dict.items() if v == nh_ip])
-
                 bank_ids = [str(x) for x in bank_ids]
-
                 if nhg_prefix in output_dict:
                     output_dict[nhg_prefix].append(nh_ip.split("@")[0])
                 else:
                     output_dict[nhg_prefix] = [nh_ip.split("@")[0]]
                 bank_dict[nh_ip.split("@")[0]] = bank_ids
-
             bank_dict = OrderedDict(sorted(bank_dict.items()))
             nhg_prefix_report = (nhg_prefix.split("|")[1])
-
             for nhip, val in bank_dict.items():
-                formatted_banks = ','.replace(',', '\n').join(bank_dict[nhip])
-                table.append([nhg_prefix_report, nhip, formatted_banks])
-
-        click.echo(tabulate(table, header, tablefmt="grid"))
-
+                displayed_banks = []
+                bank_output = ""
+                formatted_banks = (bank_dict[nhip])
+                for bankid in formatted_banks:
+                    if (len(str(bankid)) == 1):
+                        displayed_banks.append(str(bankid) + "  ")
+
+                    if (len(str(bankid)) == 2):
+                        displayed_banks.append(str(bankid) + " ")
+
+                    if (len(str(bankid)) == 3):
+                        displayed_banks.append(str(bankid))
+                for i in range (0, len(displayed_banks), 8):
+                    bank_output = bank_output + " ".join(displayed_banks[i:i+8]) + "\n" 
+                bank_output = bank_output + "\n"
+                table.append([nhg_prefix_report, nhip, bank_output])
+        click.echo(tabulate(table, header, tablefmt="simple"))
     else:
-        for nhg_prefix, alias in fg_nhg_alias.items():
-            if nhg == alias:
-                if ":" in nhg_prefix:
-                    for key in table_keys:
-                        mod_key = key.split("|")[1].split("/")[0]
-                        mod_nhg_prefix = nhg_prefix.split("/")[0]
-                        if ipaddress.ip_address(mod_key).exploded == ipaddress.ip_address(mod_nhg_prefix).exploded:
-                            t_dict = state_db.get_all(state_db.STATE_DB, key)
-                    nhg_prefix = "FG_ROUTE_TABLE|" + nhg_prefix
-                else:
-                    nhg_prefix = "FG_ROUTE_TABLE|" + nhg_prefix
-                    t_dict = state_db.get_all(state_db.STATE_DB, nhg_prefix)
-
+        header = ["FG NHG Prefix", "Next Hop", "Hash buckets"]
```

Comment:
Do we need an overridden header, seems to be the same set of fields

Reply:
got it can address the redundant header decls